content/en/infrastructure/containers/configuration.md

```diff
@@ -266,13 +266,129 @@ Follow these steps to collect the custom resources that these CRDs define:
 
 1. On the modal, under **Indexing Configuration**, select the fields you want to index from the custom resource.
 
-   {{< img src="infrastructure/containers_view/CRD_indexing_2.mp4" alt="A video of the Collecting and Indexing modal. The cursor selects three fields and clicks Enable Indexing. A success message displays." video="true">}}
-     
-   Select **Enable Indexing** to save.
+    {{< img src="infrastructure/containers_view/CRD_indexing_2.mp4" alt="A video of the Collecting and Indexing modal. The cursor selects three fields and clicks Enable Indexing. A success message displays." video="true">}}
 
-   <div class="alert alert-info">You can select a maximum of 50 fields for each resource.</div>
+#### Indexing complex types
+
+For arrays of objects, two indexing strategies are available:
+
+-   **Field**: Groups object array nested fields under each unique value of the selected field.
+-   **No field**: Flattens object array nested fields into arrays, merging all field values.
+
+##### Example: Filtering on DatadogPodAutoscaler custom resources
+
+Consider these two custom resources:
+
+{{< tabs >}}
+{{% tab "CR1" %}}
+
+```yaml
+status:
+    conditions:
+        - type: HorizontalAbleToScale
+          status: 'True'
+        - type: VerticalAbleToApply
+          status: 'False'
+```
+
+{{% /tab %}}
+{{% tab "CR2" %}}
+
+```yaml
+status:
+    conditions:
+        - type: VerticalAbleToApply
+          status: 'True'
+        - type: HorizontalAbleToScale
+          status: 'False'
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+You have the filtering possibilities on `status.conditions` based on the two indexing strategies:
+
+{{< tabs >}}
+{{% tab "Grouping by no field" %}}
+
+**Indexed fields for CR1:**
+
+```yaml
+status:
+    conditions:
+        type: [HorizontalAbleToScale, VerticalAbleToApply]
+        status: ['True', 'False']
+```
+
+**Indexed fields for CR2:**
+
+```yaml
+status:
+    conditions:
+        type: [VerticalAbleToApply, HorizontalAbleToScale]
+        status: ['True', 'False']
+```
+
+**Example queries:**
+
+**Query 1:**
+
+```text
+field#status.conditions.status:"False"
+```
+
+**Result:** Returns CR1 and CR2. Both CRs have at least one object with `status:"False"`
+
+**Query 2:**
+
+```text
+field#status.conditions.status:"False" AND field#status.conditions.type:VerticalAbleToApply
+```
+
+**Result:** Returns CR1 and CR2. Each filter is evaluated separately. At least one `status.condition` object in each custom resource matches each filter, even if it's not the same object that matches both filters.
```

Comment:
>Each filter is evaluated separately

This makes me think that the queries `field#status.conditions.status:"False"` and `field#status.conditions.type:VerticalAbleToApply` are evaluated separately in the backend which is not true.

```suggestion
**Result:** Returns CR1 and CR2. At least one `status.condition` object in each custom resource matches one of the filters - even if it's not the same object that matches both filters.
```